models/migrations/v219.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+	"time"
+
+	"code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/timeutil"
+	"xorm.io/xorm"
+)
+
+func addSyncOnCommitColForPushMirror(x *xorm.Engine) error {
+	type PushMirror struct {
+		ID         int64            `xorm:"pk autoincr"`
+		RepoID     int64            `xorm:"INDEX"`
+		Repo       *repo.Repository `xorm:"-"`
+		RemoteName string
+
+		SyncOnCommit   bool `xorm:"NOT NULL DEFAULT false"`
```

Comment:
Hmm, is there a reason not to make that `true` by default?
I mean, it _could_ potentially be a slight breaking change if some external systems are completely misimplemented, but for normal systems, this should make no difference.
I think most mirrors would like to activate this by default...

Reply:
For repos with a lot of push commits, would that bring too much sync operations? It shouldn't be an issue in most cases, though.